### ISSUE TITLE
"No video with supported format or MIME type found"

### ISSUE THREAD

**Issue by telmer-adomix** [pablouser1/ProxiTok#109]
Recently I started noticing that, at several instances and in several profiles, only some videos are properly played, while with the others the error message is showed. What videos are played or not appears to be quite random.

**To Reproduce**
Example:
1. Go to https://proxitok.pabloferreiro.es/@avvocatodellatomo (or in other istances). (It's an Italian profile about divulgation on nuclear power and energy infrastructures).
2. Click play on the videos.
3. As to now, the three videos from the top downwards (whose descriptions start, respectively, with: "Per quando vi diranno che gli...",  "#greenscreen Tra la tr4gedia...", "7 caratteristiche dell'#ambientalismo...") don't work,  the two following ones ("Risposta a @_yrion_...", "come #camminano le fonti...") work, and working and not working videos alternate thereon.

I cannot understand if it's a bug on my side, but for example cleaning the cache appears not to work.

**Comment by NoPlagiarism** [comment]
Seems working to me, from which browser and device you were getting this error?

**Comment by telmer-adomix** [comment]
I use the last version of Firefox (107.0.1) on a Windows laptop. But actually now the videos also work for me (even if buffering is still a bit longer than before)! Same on other profiles where the same message was shown. I don't know if some update to the server was progressing, or there were other "glitches".
I will update if the problem happens again.

**Comment by pablouser1** [comment]
Looks like it has to do with a new subdomain that TikTok is using... All the videos that use `v19-webapp-prime.tiktok.com` return a 403 error. I'm looking into it now

**Comment by pablouser1** [comment]
> I use the last version of Firefox (107.0.1) on a Windows laptop. But actually now the videos also work for me (even if buffering is still a bit longer than before)! Same on other profiles where the same message was shown. I don't know if some update to the server was progressing, or there were other "glitches". I will update if the problem happens again.

It also looks OK for me now... Really weird. Now all the videos that I get only use the `v16-webapp.tiktok.com` domain, there are none using `v19-webapp-prime.tiktok.com`... Maybe something on TikTok's side that ended up getting removed? If I don't see this happening again I guess I'll close the issue.

**Comment by telmer-adomix** [comment]
Sorry I didn't answer before. I checked the service during the last week and never encountered the problem again.
Judging on the names, TikTok may have been testing a new version of their interface.

**Comment by brian6932** [comment]
This issue is back

**Comment by NoPlagiarism** [comment]
@brian6932 
While it's on official instance. Try another one. 

[List](https://github.com/pablouser1/ProxiTok/wiki/Public-instances) | [Working instance](https://proxitok.esmailelbob.xyz)

**Comment by brian6932** [comment]
The issue was occurring on the first party TikTok site as well is my issue here

**Comment by NoPlagiarism** [comment]
Still, I had this issue tho. Mb FireFox problem?
What browser are you using?

**Comment by brian6932** [comment]
Firefox 109

**Comment by Andrewskiz** [comment]
I'm having the same issue on Firefox 109, I've noticed that these instances seem not to have this problem as often. I don't know if there is any difference in configuration or if it's a coincidence:  

https://proxitok.pussthecat.org/

 https://tt.vern.cc/

https://proxitok.pufe.org/

**Comment by mitomenai** [comment]
This is impacting all browsers. Firefox shows "no video" while Chrome doesn't do anything. It's broken because the video returns a 403 Forbidden response from the server. 

I've confirmed this with FIddler.

**Comment by pablouser1** [comment]
Investigating a little bit, looks like now these new urls require to have a cookie `tt_chain_token` sent, this value is sent by the server after entering `https://www.tiktok.com`. I'm trying to implement this efficiently now.

**Comment by pablouser1** [comment]
> Investigating a little bit, looks like now these new urls require to have a cookie `tt_chain_token` sent, this value is sent by the server after entering `https://www.tiktok.com`. I'm trying to implement this efficiently now.

This is now properly implemented thanks to pablouser1/TiKScraperPHP@4426ee91c13b4661fd11e1d3f4b5da33c64fbc75

`tt_chain_token` is fetched on the constructor of the Stream or Download class, allowing it to be used later. This fix is now deployed on the official instance.

**Comment by privacytime101** [comment]
> This is now properly implemented thanks to [pablouser1/TikScraperPHP@4426ee9](https://github.com/pablouser1/TikScraperPHP/commit/4426ee91c13b4661fd11e1d3f4b5da33c64fbc75)
> 
> `tt_chain_token` is fetched on the constructor of the Stream or Download class, allowing it to be used later. This fix is now deployed on the official instance.

Is there a fix coming to this repo as well? I just cloned this one and threw it in docker.

**Comment by NoPlagiarism** [comment]
> Is there a fix coming to this repo as well?

Yes, https://github.com/pablouser1/ProxiTok/commit/e8ec0ab293cb96307c644c10d50fde84176e7852

**Comment by samip5** [comment]
It seems that `v19-webapp-prime.tiktok.com` returns forbidden.

Eg. `https://v19-webapp-prime.tiktok.com/video/tos/alisg/tos-alisg-pve-0037c001/oQlbXTwUh5zAzopfaTAGSxTGF2dAICRGIocStr/?a=1988&ch=0&cr=0&dr=0&lr=tiktok_m&cd=0%7C0%7C1%7C0&cv=1&br=3798&bt=1899&cs=0&ds=3&ft=4fUEKMR58Zmo0jV7l64jVKa_JpWrKsdm&mime_type=video_mp4&qs=0&rc=aTk8NDg6ZzNmNmg6ZTo8ZUBpamxoazg6ZnZuaDMzODczNEAuNC42MWMvNmExYTJjNGJjYSNfamZicjRvM2FgLS1kMS1zcw%3D%3D&btag=80000&expire=1675869561&l=202302080910495D21C8D4D775EA16AAD0&ply_type=2&policy=2&signature=9408fde064eb3caacbbd678990f09157&tk=tt_chain_token`

Results in forbidden from Varnish Cache Server:

```
<html>
  <head>
    <title>403 Forbidden</title>
  </head>
  <body>
    <h1>Error 403 Forbidden</h1>
    <p>Forbidden</p>
    <h3>Error 54113</h3>
    <p>Details: cache-hel1410030-HEL 1675847461 1322291421</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```

**Comment by cron410** [comment]
For me at least, it seems like the non-watermarked video is able to be downloaded, but the watermarked video downloads a 400byte plaintext file containing:

```
Access Denied
You don't have permission to access "http://v16-webapp-prime.tiktok.com/video/tos/alisg/tos-alisg-pve-**********/****************************************/?" on this server.
Reference #18.b*********************.1**************** 
```

Would it be possible to add a setting that allows us to set the default video type for the player?

**Comment by cron410** [comment]
It seems #111 is back.

**Comment by cunlem** [comment]
> For me at least, it seems like the non-watermarked video is able to be downloaded

Yes I've noticed this too. Maybe it would make sense to switch the embedded video source to non-watermarked version for the time being?

**Comment by cron410** [comment]
It's been working fine for me recently. My docker container is set to auto update every night so I have the latest version

**Comment by igalic** [comment]
I have seen this error as transitory, on flaky networks, for instance

**Comment by igalic** [comment]
reload often fixes it

**Comment by batvin3211** [comment]
i am new to this project. i set up proxitok under docker and i could not play a video due to MiME issue. when i examined the video URL i found out if i remove the "us" from the "https%3A%2F%2Fv16-webapp-prime.us.tiktok.com" part of the URL it plays "https://proxitok.ascosilinux.com/stream?url=https%3A%2F%2Fv16-webapp-prime.tiktok.com%2Fvideo%2Ftos%2Fuseast2a%2Ftos-useast2a-ve-0068c001-euttp%2FoA9CPAbsPGIgSvRgEIQsQ0eDTeJOKlw5nIDFBW%2F%3Fa%3D1988%26ch%3D0%26cr%3D3%26dr%3D0%26lr%3Dunwatermarked%26cd%3D0%257C0%257C0%257C3%26cv%3D1%26br%3D1888%26bt%3D944%26bti%3DODszNWYuMDE6%26cs%3D0%26ds%3D6%26ft%3D_rKBMBnZq8Zmoe2dXQ_vjzZGsAhLrus%26mime_type%3Dvideo_mp4%26qs%3D0%26rc%3DZDc7NDVnNzw7Ojg8NTQ8OEBpajpzbzo6Zmw4bjMzZjczM0BeNDU2XzMtXmIxLTQtMDIxYSMxaGZkcjQwcmBgLS1kMWNzcw%253D%253D%26btag%3De00008000%26expire%3D1696659761%26l%3D20231007002201389586C01C41580E135D%26ply_type%3D2%26policy%3D2%26signature%3D90ab0d9a60230c4eab273fd3434ce018%26tk%3Dtt_chain_token"

**Comment by misterpyrrhuloxia** [comment]
@pablouser1, I seem to be experiencing this issue now too, but _only_ with my instance of Proxitok. It was never an issue for me on my own instance until the last week or two. When opening any tiktok via my instance, the page loads but when I click on the video's play button, as well as the audio track's play button, the button becomes grayed out and neither the video nor audio plays. When I open any videos in a Firefox private tab with my instance of Proxitok, I get the `No video with supported format and MIME type found`. No amount of refreshing has solved this for me. I do not experience the same issue with any tiktoks on most of the rest of the public instances of Proxitok. 
I host my instance from my Windows 11 desktop pc via Docker using your [docker-compose.yml](https://github.com/pablouser1/ProxiTok/blob/master/docker-compose.yml). No amount of re-pulling and redeploying the container has helped.

**Comment by JakeShirley** [comment]
I just set ProxiTok up locally and am experiencing this. Hosting it on my unraid box with docker compose, using Firefox 122.0.

Example URL: 
https://v16-webapp-prime.us.tiktok.com/video/tos/useast8/tos-useast8-ve-0068c004-tx2/owFLvDYNIAqR1TgIwIYEhXeJFQejAPFlLwCVqf/?a=1988&ch=0&cr=3&dr=0&lr=unwatermarked&cd=0%7C0%7C0%7C&cv=1&br=1676&bt=838&bti=ODszNWYuMDE6&cs=0&ds=6&ft=4KJMyMzm8Zmo0csDW94jV0WrQpWrKsd.&mime_type=video_mp4&qs=0&rc=aGVnZWRlZThkNDc8ZTM2ZEBpajxkZHI5cjc4cDMzaTczNEBjXi9jMDNfNjMxNDFeMGBjYSNua2phMmRjc2xgLS1kMTJzcw%3D%3D&btag=e00088000&expire=1706852228&l=20240201233623DEF95E8C9BD67F021F90&ply_type=2&policy=2&signature=816f23ece405a8b3a2d89d37a54d541a&tk=tt_chain_token

**Comment by SiskoUrso** [comment]
I never was able to resolve this issue after setting up and keeping updated for a few months hoping for some change that might fix it but no go. 

Hoping there is a resolution to this.

**Comment by FoxxMD** [comment]
Also experiencing this issue on that [`latest` docker image](https://github.com/pablouser1/ProxiTok/pkgs/container/proxitok/170993116?tag=latest) when hosting myself

**Comment by pablouser1** [comment]
Should be fixed with #224
